### ISSUE TITLE
./isosurface/computeMarchingCubes requires libpng12

### ISSUE THREAD

**Issue by janbechtold** [devlearning-gt/3DShapeGen#1]
Hi,
thanks for providing the code to your paper.
I wanted to generate meshes from the sdf values using your pipeline `eval.py` but run into the error:
```bash
(sdf_net)` user@host SDFNet: python eval.py            
Loading model from 3DShapeGen/out/train_overfit_test/best_model_iou.pth.tar
Loading data...
  0%|                                                                  | 0/1 [00:00<?, ?it/s]
./isosurface/computeMarchingCubes: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
```
in the environment.yaml another libpng version is specified: `  - libpng=1.6.37=hed695b0_0`
I tried to downgrade the libpng version to 12 in the conda environment. This did not work.
Could you please provide (or point me to) the codebase s.t. I can build MarchingCubes myself with libpng16 or could you update this repo with a new build that uses the version from the environment?

Thanks a lot in advance.

**Comment by sstojanov** [comment]
Hi,

Thank you for reaching out! This seems to be an issue with libpng12 being available for Ubuntu 16.04 but not 18.04. I had this issue as well when I upgraded to a system with 18.04. For me it worked to make a symbolic link for libpng12 to point to libpng16. The command would look something like the following

`sudo ln -s -T /usr/lib/x86_64-linux-gnu/libpng16.so /usr/lib/x86_64-linux-gnu/libpng12.so.0`

We are borrowing the binaries directly from the DISN paper. We (currently) don't know exactly where the source for these is from, so we don't know how to do a build with libpng16.

Alternatively, the [marching cubes implementation from skimage](https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.marching_cubes_lewiner) can be used to obtain a mesh.

Please follow up with whether this helps out or not. Thanks!

**Comment by janbechtold** [comment]
Hi,

thanks for your quick response.
Creating a symbolic link in the conda environment solved the issue.
